litestar/_openapi/schema_generation/schema.py

```diff
@@ -669,12 +670,16 @@ def for_struct_class(self, annotation: type[Struct], dto_for: ForType | None) ->
         Returns:
             A schema instance.
         """
+
+        def _is_field_requiured(field: FieldInfo) -> bool:
+            return field.required or field.default_factory is Empty
+
         return Schema(
             required=sorted(
                 [
                     field.encode_name
                     for field in msgspec_struct_fields(annotation)
-                    if field.required and not is_optional_union(field.type)
+                    if _is_field_requiured(field=field) and not is_optional_union(field.type)
```

Comment:
```suggestion
                    if _is_field_required(field=field) and not is_optional_union(field.type)
```

Reply:
This indicates also that this is not tested. This should have raised.